lib/token.js

```diff
@@ -1,14 +1,41 @@
 const BN = require('ethjs').BN
 const util = require('./util')
 
+// bn.js requires extra validation to safely use
+// See here: https://github.com/indutny/bn.js/issues/151
+function _isInvalidBnInput (input) {
+  return (
+    typeof input === 'string' &&
+    (
+      input.startsWith('0x') ||
+      Number.isNaN(parseInt(input))
```

Comment:
Whoops - this string is parsed as hex.
```suggestion
      Number.isNaN(parseInt(input, 16))
```

Reply:
Actually no - this is expected to be a hex string in one place, but a decimal string in another.

Reply:
Fixed here: 7b12972da371e7fee8271e29c2179a8a9376fa47
Added additional tests for this as well.